course_catalog/tasks_helpers.py

```diff
@@ -60,33 +66,31 @@ def parse_mitx_json_data(course_data):
             except Course.DoesNotExist:
                 course_instance = None
 
-            try:
-                year = int(course_run_key[-4:])
-            except ValueError:
-                year = datetime.strptime(course_run.get("start"), "%Y-%m-%dT%H:%M:%S.%fZ").year
+            year, semester = get_year_and_semester(course_run, course_run_key)
 
             course_fields = {
                 "course_id": course_run_key,
                 "title": course_run.get("title"),
                 "short_description": course_run.get("short_description"),
                 "full_description": course_run.get("full_description"),
                 "level": course_run.get("level_type"),
-                "semester": semester_mapping.get(course_run_key[-6:-4], None),
+                "semester": semester,
                 "language": course_run.get("content_language"),
                 "platform": PlatformType.mitx.value,
                 "year": year,
                 "start_date": course_run.get("start"),
                 "end_date": course_run.get("end"),
                 "enrollment_start": course_run.get("enrollment_start"),
                 "enrollment_end": course_run.get("enrollment_end"),
-                "image_src": course_run.get("image").get("src"),
-                "image_description": course_run.get("image").get("description"),
+                "image_src": (course_run.get("image") or {}).get("src"),
+                "image_description": (course_run.get("image") or {}).get("description"),
                 "last_modified": max_modified,
                 "raw_json": json.dumps(course_data),
             }
 
             course_serializer = CourseSerializer(data=course_fields, instance=course_instance)
             if not course_serializer.is_valid():
+                # print(course_serializer.errors)
```

Comment:
Would be useful to have logging here, something like:
```python
log.exception("Course {} is not valid: {}".format(course_run_key, course_serializer.errors))
```

Reply:
Added.